src/OSPSuite.R/RRegister.cs

```diff
@@ -41,7 +46,13 @@ public override void RegisterInContainer(IContainer container)
          container.Register<SimulationBatch, SimulationBatch>();
          container.Register<ISimulationBatchFactory, SimulationBatchFactory>();
          container.Register<IObjectTypeResolver, RObjectTypeResolver>();
-
+         container.Register<HeavyWorkManager, HeavyWorkManager>();
+         container.Register<IHeavyWorkManager, HeavyWorkManager>();
+         container.Register<IHeavyWorkPresenterFactory, HeavyWorkPresenterFactory>();
+         container.Register<HeavyWorkPresenter, HeavyWorkPresenter>();
+         container.Register<HeavyWorkCancellablePresenter, HeavyWorkCancellablePresenter>();
+         container.Register<IHeavyWorkCancellableView, HeavyWorkCancellableView>();
```

Comment:
After the changes to the `HeavyWorkPresenterFactory` remove these registrations and remove the project references. If there are IoC failures we'll need to find another way to solve that failure because we can't add the ProjectReferences to support this.